File: leetcode-in-golang/algo/sort/performance.go
package sort

import (
	"fmt"
	"leetcode/util"
	"time"
)

func timeCost() func() {
	start := time.Now()
	return func() {
		tc := time.Since(start)
		fmt.Printf("time cost = %v\n", tc)
	}
}

type Algo struct {
	Name     string
	SortFunc func([]int)
}

func AlgoRandomPerformance(algos []Algo, size, l, r int) {
	arr := util.RandomNums(size, l, r)
	nums := make([]int, len(arr))
	for _, algo := range algos {
		copy(nums, arr)
		func() {
			fmt.Printf("%s ", algo.Name)
			defer timeCost()()
			algo.SortFunc(nums)
		}()
	}
}

func AlgoNearlyOrderPerformance(algos []Algo, size, swap int) {
	arr := util.RandomNearlyOrderNums(size, swap)
	nums := make([]int, len(arr))
	for _, algo := range algos {
		copy(nums, arr)
		func() {
			fmt.Printf("%s ", algo.Name)
			defer timeCost()()
			algo.SortFunc(nums)
		}()
	}
}

File: leetcode-in-python/algo/sort/performance.py
from typing import List, Callable
import helper
import time
from insert_sort import insert_sort
from select_sort import select_sort
from quick_sort import quick_sort


def algo_performance(algos: List[Callable[[List[int]], None]], size, l, r):
    arr = helper.random_nums(size, l, r)
    for fn in algos:
        nums = arr[:]
        start = time.time()
        fn(nums)
        end = time.time() - start
        print(f"{fn.__name__}: {end} s")


algo_performance([select_sort, insert_sort, quick_sort], 1000, 0, 1)
